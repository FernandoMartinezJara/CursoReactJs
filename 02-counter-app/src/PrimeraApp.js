import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {

    const devs = {
        nombre: 'Fernando',
        edad: 66
    }

   return <>
        <h1>{ saludo }</h1>
        {/* <pre>{JSON.stringify(devs, null, 3)}</pre> */}
        <p>{ subtitulo }</p>
    </>;
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
    subtitulo: 'Esta es mi primera app'
}

export default PrimeraApp;