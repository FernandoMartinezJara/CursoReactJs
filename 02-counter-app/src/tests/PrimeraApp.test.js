
import {shallow} from 'enzyme';
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp />', () => {

    //React library test
    // test('debe mostrar el mensaje CounterApp', () => {
    //     const saludo = 'CounterApp';
    //     const { getByText } = render(<PrimeraApp saludo={'CounterApp'} />);

    //     expect(getByText(saludo)).toBeInTheDocument();
        
    // });

    test('dene de mostrar <PrimeraApp/>', () => {

        const saludo = 'CounterApp'
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe mostrar el subitulo enviado por props ', () => {
        const saludo = 'Hola, Soy goku';
        const subtitulo = 'Soy un subtitulo!';
        const wrapper = shallow(
            <PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe(subtitulo);
        
    })
    
    


    
    
})
