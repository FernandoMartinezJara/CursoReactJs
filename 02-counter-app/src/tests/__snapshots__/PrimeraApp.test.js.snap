// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> dene de mostrar <PrimeraApp/> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    Esta es mi primera app
  </p>
</Fragment>
`;
